TestMauidotnet8/MainPage.xaml.cs

```diff
@@ -24,12 +24,7 @@ private void OnCounterClicked(object sender, EventArgs e)
 
             SemanticScreenReader.Announce(CounterBtn.Text);
 
-            MyViewModel vm = new()
-            {
-                Text = "My modified text"
-            };
-
-            popupService.ShowPopup(vm);
+            popupService.ShowPopup<MyViewModel>();
```

Comment:
In your `ContentPage`, where you do not have access to an instance of `MyViewModel`, you don't need to (and shouldn't) instantiate `MyViewModel` to pass it in as a parameter.